File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        Map<Integer, Integer> m = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int difference = target - nums[i];
            if (m.containsKey(difference)) {
                res[0] = m.get(difference);
                res[1] = i;
                break;
            } else {
                m.put(nums[i], i);
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        m = len(nums) // 2
        r = len(nums) - 1

        while l <= r:
            item = nums[m]
            if (target == item):
                return m
            if (target < item):
                r = m - 1
            if (target > item):
                l = m + 1  
            m = (r + l) // 2
        
        return -1